### DISCUSSION TITLE
Any way to know if navigation occured due to the app calling the page api

### DISCUSSION THREAD

**Discussion by dhrumil83** [microsoft/Microsoft-Fabric-workload-development-sample#137] [Q&A] [answered]
[Page open](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/src/controller/SampleWorkloadController.ts#L138) would trigger [onNavigate](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/src/index.ui.tsx#L15)

Any way to know when onNavigate is triggered that it was an application-initiated page open?
Ideally, we would want different behavior when an app-initiated page opens (create an item and go to it) compared to when the user clicks on a workload item from the workspace page.

Also, what does passing OpenMode(Append vs ReplaceAll) in the Page API do? 
I was looking for that info to appear onNavigate, but it does not.

**Comment by anitamayorov** [comment, marked as answer]
Hi @dhrumil83,

The onNavigate params (WorkloadNavigation) have no 'source' only the target.
You can try and look at the createExperience.action (e.g., "open.createSampleWorkload") which is triggered only upon create and see if you can define different paths to load there if you need to handle different scenarios, then you can look at the targetUrl - it will be different depends on the path provided by create action handling or the item's manifest ('editor'.'path').

**Reply by dhrumil83** [reply]
thanks for the targetUrl tip.
